### ISSUE TITLE
NoWhitespaceBefore and empty for loop conditions

### ISSUE THREAD

**Issue by cushon** [checkstyle/checkstyle#5058]
The Google Java style guide requires horizontal whitespace after `;`: https://google.github.io/styleguide/javaguide.html#s4.6.2-horizontal-whitespace. In for loops with an empty condition, this means that there should be whitespace after the `;` that terminates the initializer and before the `;` that terminates the condition. [NoWhitespaceBefore](http://checkstyle.sourceforge.net/config_whitespace.html#NoWhitespaceBefore) currently emits an unwanted error in this case.

```java
class EmptyForLoop {
  public static void f() {
    for (; ; ) {} // these spaces are required by Google Style
  }
}
```

```xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE module PUBLIC "-//Puppy Crawl//DTD Check Configuration 1.3//EN" "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="NoWhitespaceBefore">
      <property name="tokens" value="SEMI"/>
      <property name="allowLineBreaks" value="true"/>
    </module>
  </module>
</module>
```

```bash
$ java -jar checkstyle-8.2-all.jar -c checks.xml Test.java
...
[ERROR] Test.java:3:11: ';' is preceded with whitespace. [NoWhitespaceBefore]
```

**Comment by romani** [comment]
looks like you already created such issue - #4294

**Comment by cushon** [comment]
Nope, that's about ParenPad, this is about NoWhitespaceBefore. The other bug uses `for (; ; )` as an example to point out that ParenPad doesn't warn about `)` being preceeded by whitespace in that case.

**Comment by romani** [comment]
Fix is merged.

@cushon , thanks a lot for fix.
